### PR TITLE
Road to rails 7: change form_with to local by default

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

In rails 6.1 they changed the standard of form_with defaulting to `remote: true` to `local: false` to generate an ajax form. but only for forms, not for `link_to` or `button_to`, which still use `remote: true` (a fact that was so frustrating that I might submit a patch to rails about it, haha). This makes changes around form_with, and explicitly declares in button_to and link_to functions. (Also one minor test hack to make some ajax more reliable.)

This pull request makes the following changes:
* turn on the default config in the framework default
* change a bunch of `remote: true` to `local: false`-es on forms
* explicitly declare `remote: true/false` on button_to and link_tos

no view changes

It relates to the following issue #s: 
* Bumps #2462 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: app/views/patients/_fulfillment.html.erb [hunk 1]
```diff
@@ -13,8 +13,7 @@
 
   <%= bootstrap_form_with model: patient,
                           html: { id: 'pledge_fulfillment_form' },
-                          method: 'patch',
-                          remote: true do |f| %>
+                          local: false do |f| %>
```
Comment (lomky) [inline app/views/patients/_fulfillment.html.erb:16]: Why'd we lose the `patch` here?

File: app/views/dashboards/_search_form.html.erb [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <%= t 'dashboard.search.header' %>
   </h1>
 
-  <%= bootstrap_form_with url: '/search', layout: :inline do |p| %>
+  <%= bootstrap_form_with url: '/search', layout: :inline, local: false do |p| %>
```
Comment (lomky) [inline app/views/dashboards/_search_form.html.erb:6]: is this one a change, or just an implicit made explicit?

Comment (colinxfleming) [reply app/views/dashboards/_search_form.html.erb:6]: yep, this is an implicit made explicit!

File: config/initializers/new_framework_defaults_6_1.rb [hunk 3]
```diff
@@ -45,7 +45,7 @@
 Rails.application.config.active_record.legacy_connection_handling = false
 
 # Make `form_with` generate non-remote forms by default.
-# Rails.application.config.action_view.form_with_generates_remote_forms = false  # HARD ONE
```
Comment (lomky) [inline config/initializers/new_framework_defaults_6_1.rb:48]: Nit: kill the comment?

**Review by lomky — APPROVED** [review]
🎉

**Pull request merged**